extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/StaticResourcesProcessor.java

```diff
@@ -75,7 +75,15 @@ public void nativeImageResource(Optional<StaticResourcesBuildItem> staticResourc
                     // TODO: do we perhaps want to register the whole directory?
                     continue;
                 }
-                String metaInfResourcesPath = StaticResourcesRecorder.META_INF_RESOURCES + entry.getPath();
+                String path = entry.getPath();
+                int index = path.lastIndexOf("/index.html");
```

Comment:
> Is index.html the only default page name supported in JVM mode? What about index.htm for example

Good point! It's actually configurable, so I'll use the config property

> Thinking out loud: Why not registering all directory paths leading up to a file?

Because no one really needs them AFAIK.

Reply:
> Good point! It's actually configurable, so I'll use the config property

Actually the property is build time... So there is not much we can do other than move the property to build time (which we would be a breaking change).

Reply:
Or reconsider the

> Because no one really needs them AFAIK.

part :)

If I get it right that would allow the users to still set the property at runtime and have it work for any file.

Reply:
That still would not be enough

Reply:
Could you elaborate on why it would not be enough?

Also at the light of this, what's the plan? Merge this PR and at a later point move the property to build time?

If we are going to use fixed names perhaps we should use a set of potential default page names instead of supporting only `index.html`.

Reply:
> Also at the light of this, what's the plan? Merge this PR and at a later point move the property to build time?

The plan is to think about moving it build if someone actually complains about this.

> If we are going to use fixed names perhaps we should use a set of potential default page names instead of supporting only index.html.

That involves changes in Vertx as well and given how niche this feature seems to be (the bug was just now reported and it's been there since day 1), I don't think it warrants the effort (or put another way, I am certainly not going to be the one doing that work).

Reply:
Should we just do it for all directories in the `META-INF/resources` directory? I don't think it will hurt.

Reply:
You mean register all directories under `META-INF/resources`? We could do it, but what would that buy us?

Reply:
Well, my understanding is that you were trying to cherry pick the directories with an index resource, which would be hard given you don't know the name of the index resource for sure at this point.
So my question is: I don't think it would cost much to register all directories in `META-INF/resources` and simplify the problem.
And I don't foresee any bad side effects.

Or did I miss something?

Reply:
Oh, now I see what you mean.

I support we could do that, it probably won't cost much of anything when it comes to native image size.
A follow up question to that approach would be whether we went to do this recursively or not.

Reply:
Yes, I think you need to do that recursively given an `index.html` can be nested.

Reply:
FWIW that was what I meant as well. +1

Reply:
Should be fixed now